kratos/utilities/global_pointer_utilities.h

```diff
@@ -363,16 +362,20 @@ class GlobalPointerUtilities
     ///@name Member Variables
     ///@{
 
-    template< class TIteratorType >
-    static bool IteratorIsLocal(TIteratorType& it, const int CurrentRank)
```

Comment:
why do you change this?
Looks correct to me

Reply:
Is correct. In fact this is the problem.
The function is correct and is so generic that everything can enter there. The problem comes when you have a let's say a node iterator

You would expect that to enter in: 

`static bool IteratorIsLocal(ModelPart::NodesContainerType::iterator& it, const int CurrentRank)`

which checks if the node belongs to the local partition or nor, but in fact you may have:

`const ModelPart::NodesContainerType::iterator &`
`ModelPart::NodesContainerType::const_iterator &`
`const ModelPart::NodesContainerType::const iterator &`

And all of them will by default resolve to use the generic variant, which will casue all the nodes to be treated as if they were from the local partition, whereas they are or not.

By explicitly typifying the function, at least, we can cause a compilation error if its not possible to call it, which will point to the error, instead of returining `True` by default.

Reply:
ok makes sense :)

Reply:
nasty one (i'll take the blame for this)